imap/jmap_backup.c

```diff
@@ -321,12 +321,16 @@ static void restore_resource_cb(const char *resource __attribute__((unused)),
         goto done;
     }
 
-    if (restore->msgno_torecreate) {
-        recreatemsg = message_new_from_mailbox(mailbox, restore->msgno_torecreate);
+    if (restore->uid_torecreate) {
+        struct index_record record;
+        mailbox_find_index_record(mailbox, restore->uid_torecreate, &record);
```

Comment:
Here and elsewhere we don't check the return value of mailbox_find_index_record. Should we? Or is the point that if we got here and it does fail, something is awful messed up anyway and crashing later is okay?

Reply:
Yeah, we need to check that it succeeds.  It could be that the UID we're planning to work with got removed by a repack.

Reply:
Oh actually there's something worse too in not checking that. We had an issue awhile back because of this bit of mailbox_find_index_record:
```
    /* wasn't the actual record? */
    if (record->uid != uid) return IMAP_NOTFOUND;

```
So it might fill in `&record` and then return saying "I didn't find it" and then later code could work on the wrong record